### DISCUSSION TITLE
Switch to new workspace via wezterm cli

### DISCUSSION THREAD

**Discussion by faeb5** [wezterm/wezterm#3534] [Q&A] [answered]
Hi, I have a script shell script that let's me choose a directory from a list of directories. It then checks if there is a workspace for the chosen directory and if not creates a new workspace for it. However, in any case I want to immediately switch to the newly (or not) created workspace. Can I achieve this using `wezterm cli`? I've already tried `wezterm cli activate-pane` and `wezterm cli activate-tab` but those commands seem to do nothing, which makes sense because I'm still in the default workspace. Any ideas? Here is the script so far.

```bash
#!/bin/sh

if [ "$#" -eq 1 ]; then
  project="$1"
else
  project=`fdfind -L --min-depth=1 --max-depth=1 -t directory . ~/work ~/.config | fzf`
fi

[ -z "$project" ] && exit 0

workspace=$(basename "$project")
workspace_dir="$project"

if wezterm cli list | awk '{ print $4 }' | grep -iq "$workspace"; then
    echo "workspace already exists"
    # TODO switch to new workspace
else
    echo "workspace does not exist"
    wezterm cli spawn --new-window --cwd "$workspace_dir" --workspace "$workspace"
    # TODO switch to new workspace
fi
```

**Comment by wez** [comment, marked as answer]
There isn't a built-in way to do this (please file an enhancement issue!), but take a look at:
* https://github.com/wez/wezterm/issues/2979#issuecomment-1447519267
for a way to pass data to the gui from a pane running inside wezterm

**Comment by faeb5** [comment]
Thanks, will do!

**Reply by honeyspoon** [reply]
I want to build the same thing
did you find a way?

**Reply by faeb5** [reply]
Hey, I haven't tried since, unfortunately.
